### PR TITLE
Adding a Docker Image to this Repo :)

### PR DESCRIPTION
**Pull request by GlassedSilver**
- [x] **Initial Image with dependencies installed**
- [x] **RDP-based Web UI rather than VNC**, although you could hook into this with VNC as well I think, the base image is [Linuxserver.io's webtops](https://www.linuxserver.io/blog/2021-05-05-meet-webtops-a-linux-desktop-environment-in-your-browser). :) (in the Ubuntu Jammy flavor, since the XFCE branch is still on Focal, and we really want Jammy we just manually install XFCE in the docker build ourselves)
- [x] All of the suite's tools can be run through the terminal or through dedicated **dock entries!** :)
- [x] **Add `docker-compose.yml`**
- [x] **Updated README**, also I allowed myself to structure it a bit differently: so the reader is caught by a screenshot, directly followed by the pitch and then the note about requirements and Docker is added along with two screenshots of the Docker desktop experience (style, etc...)
- [x] Weekly Docker builds and builds on code change. **Build time is superfast** and weekly builds help keep the base OS up to date.
- [ ] **Redo Docker building**, refer to my suggestion in Discord. (weekly from a new branch called `release` and as check for PR and pushes from branch `master`?)

**Absolutely DO NOT EXPOSE THIS TO THE INTERNET at least not directly without VPN. This runs in passwordless sudo.**

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+on:
+  schedule:
+    - cron: '41 11 * * 3'
+  push:
+    branches: [ master ]
+    # Publish semver tags as releases.
+    tags: [ 'v*.*.*' ]
+  pull_request:
+    branches: [ master ]
```
[the remark this answers is not in the document]
Comment (ThePromidius) [reply .github/workflows/docker-publish.yml:16]: From what i've seen you build the image on PRs but never use it?
What is the point of this?
Just trying to learn.

[the remark this answers is not in the document]
Comment (GlassedSilver) [reply .github/workflows/docker-publish.yml:16]: AFAIK it serves as test, but honestly I just didn't edit the default template much. I only edited the time for the cron-based schedule to run around the time Linuxserver.io creates their base images.

[the remark this answers is not in the document]
Comment (ThePromidius) [reply .github/workflows/docker-publish.yml:16]: Oh ok.
Honestly I'd like to have more control over the action. If it's only the Scheduler what was added a more custom action could be used. That way i can have dev/testing branches get built. Just like i did in manga-tagger
I also want to add some tests that run before the docker image builds.

[the remark this answers is not in the document]
Comment (GlassedSilver) [reply .github/workflows/docker-publish.yml:16]: No, the scheduler and the other bits were all there, I just modified the timing of the scheduler. Well you can change the workflow to include those tests, I haven't dabbled with that yet, so I won't be of much help there. :/

**Comment by ThePromidius** [comment]
I tried both building the image on my own and pulling your image. Seems like the custom config you did setup did not apply.

**Comment by GlassedSilver** [comment]
> I tried both building the image on my own and pulling your image. Seems like the custom config you did setup did not apply. Any idea?

Will look into it, in the meantime you can try running without mapping the config folder. That way it will not be persistent, but it only affects the desktop environment anyway. Maybe I'll make that nonpersistent by default, since the customization of the DE isn't necessary anyhow and any possible persistent data for settings and such you might want to perma-save in your app can be mapped anywhere else then :)

**Comment by ThePromidius** [comment]
I figured it out. Not sure if its really the expected behavior or just me that i misunderstood how it worked.

If i don't map the config volume (i comment it out) it will use your settings.
However if i map it to whatever folder, there won't be any settings applied.

I think of 2 solutions. Either copy the config to the mapped volume or to remove the config map setting from the docker-compose. Afterall the tool is expected to be used as a portable tool

**Comment by ThePromidius** [comment]
I'm merging this into a branch. i want to work on it.
Let me know if you have a solution to the problem above

**Pull request merged**